modules/calcite/src/main/java/org/apache/ignite/internal/processors/query/calcite/rel/AbstractIndexScan.java

```diff
@@ -92,14 +98,31 @@ public String indexName() {
         return idxName;
     }
 
+    /** */
+    public boolean isInlineScan() {
+        IgniteTable tbl = table.unwrap(IgniteTable.class);
+        IgniteIndex idx = tbl.getIndex(idxName);
+
+        if (idx != null)
+            return idx.isInlineScanPossible(requiredColumns);
+
+        return false;
+    }
+
     /** {@inheritDoc} */
     @Override public RelOptCost computeSelfCost(RelOptPlanner planner, RelMetadataQuery mq) {
         double rows = table.getRowCount();
 
         double cost;
 
+        IgniteTable tbl = table.unwrap(IgniteTable.class);
+        IgniteIndex idx = tbl.getIndex(idxName);
+
+        double inlineReward = (idx != null && isInlineScan()) ?
```

Comment:
idx != null - redundant check

Reply:
also micro optimization : isInlineScan() can be renamed and return index, no additional unwraps are raised in such a case, but its up to you.

Reply:
`isInlineScan` is used also for:
# tests
# explain plan
So such an optimization is not good for code readability

Reply:
> idx != null - redundant check

In case of some concurrent index modification it will be not redundant.